students/johnpharmd/lesson06/mailroomtest.py

```diff
@@ -0,0 +1,87 @@
+import unittest
+import mailroom4
+import os
+import datetime
+
+
+class MailroomTest(unittest.TestCase):
+    def setUp(self):
+        self.donors_amts = mailroom4.get_donors_amts()
+
+    def test_donors(self):
+        for donor in self.donors_amts:
+            if '-' not in donor:
+                assert donor.isalpha()
+            else:
+                assert donor.replace('-', '').isalpha()
+
+    def test_titles(self):
+        for donor in self.donors_amts:
+            assert self.donors_amts[donor]['title'] in ('Prof.',
+                                                        'Dr.', 'Ms.', 'Mr.')
+
+    def test_donations(self):
+        for donor in self.donors_amts:
+            assert str(self.donors_amts[donor]['donations']).isdigit()
+
+    def test_num_of_donations(self):
+        for donor in self.donors_amts:
+            assert str(self.donors_amts[donor]['num_of_donations']).isdigit()
+
+    def test_add_donor(self):
+        mailroom4.add_donor('Anderson', 'Mr.')
+        try:
+            assert 'Anderson' in self.donors_amts
+        except KeyError:
+            print('"Anderson" not in donors_amts.')
```

Comment:
why catch KeyError here? shouldn't the test fail right away if Anderson didn't get added?